packages/test/src/anvil.ts

```diff
@@ -302,20 +304,25 @@ let workerInstances = 0;
 export const spawnAnvil = async (
   args: AnvilArgs,
   index = workerInstances++,
+  testInfo?: TestInfo,
```

Comment:
IMO most versatile would be to expect 2 callbacks here: `onMessage` and `onError` and simply call each of them in place of your changes

Reply:
So the implementation is not only bound to playwright (or playwright-like `testInfo`) and you can do many more things than attaching logs

Reply:
yes indeed I broke the layer